### PR TITLE
Fixed ordering in adding pending picks to PF

### PR DESCRIPTION
**Pull request by dgquintas**
In the vein of https://github.com/grpc/grpc/pull/15947, attempt to address https://github.com/grpc/grpc/issues/15933

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by markdroth** [comment]
LGTM.  Can we add a test, similar to what we added for RR?

**Comment by dgquintas** [comment]
Yes. I'm still waiting to see if this fixes the reported nodejs issue.

**Comment by josephharrington** [comment]
@dgquintas, thanks for waiting while we verified this. Unfortunately it looks like this does not fix the issue. I added some more details in a comment on the original issue that will hopefully shed some light on the problem.

**Comment by markdroth** [comment]
Even if it doesn't fix the original issue, there's pretty clearly still a bug here, and this PR is the right fix.  We should add a test similar to what we added in the RR version and get this merged.

**Comment by dgquintas** [comment]
Test added. PTAL.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++

 -------------- SHRINK --------------
  [ = ]       0 [None]      -8  -0.0%

  [ = ]       0 TOTAL       -8  -0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

File: test/cpp/end2end/client_lb_end2end_test.cc [hunk 1]
```diff
@@ -353,6 +353,23 @@ TEST_F(ClientLbEnd2endTest, PickFirst) {
   EXPECT_EQ("pick_first", channel->GetLoadBalancingPolicyName());
 }
 
+TEST_F(ClientLbEnd2endTest, PickFirstProcessPending) {
+  StartServers(1);                  // Single server
+  auto channel = BuildChannel("");  // test that pick first is the default.
```
Comment (AspirinSJL) [inline test/cpp/end2end/client_lb_end2end_test.cc]: We should use "pick_first" here and below. The LB policy default choice is tested in PickFirst above. Other tests in this file should specify the LB policy to use.

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,952,718      Total (<)      1,952,734

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,663,575      Total (<)     10,663,579

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

File: test/cpp/end2end/client_lb_end2end_test.cc (hunk 1 above)
Comment (dgquintas) [reply test/cpp/end2end/client_lb_end2end_test.cc]: Yeah, I was being consistent with the way we do it in other tests. I suggest we do that on its own separate (and trivial) PR.

**Comment by dgquintas** [comment]
Issues: #16230 #16003

**Pull request merged**